src/Aspire.Hosting/Postgres/PostgresBuilderExtensions.cs

```diff
@@ -19,20 +19,39 @@ public static class PostgresBuilderExtensions
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <param name="port">The host port used when launching the container. If null a random port will be assigned.</param>
     /// <param name="password">The administrator password used for the container during local development. If null a random password will be generated.</param>
+    /// <param name="port">The host port used when launching the container. If null a random port will be assigned.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistributedApplicationBuilder builder, string name, int? port = null, string? password = null)
+    public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistributedApplicationBuilder builder, string name, string? password = null, int? port = null)
     {
         var postgresServer = new PostgresServerResource(name, password);
+        return builder.AddPostgres(postgresServer, port);
+    }
+
+    /// <summary>
+    /// Adds a PostgreSQL resource to the application model. A container is used for local development. This version the package defaults to the 16.2 tag of the postgres container image
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="passwordParameter">The parameter used to provide the password for the PostgreSQL resource.</param>
+    /// <param name="port">The host port used when launching the container. If null a random port will be assigned.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistributedApplicationBuilder builder, string name, IResourceBuilder<ParameterResource> passwordParameter, int? port = null)
```

Comment:
Once we decide on a pattern for password, my next change would be to add support for username parameters. The final API shape would look like this:

```C#
public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistributedApplicationBuilder builder, string name, string? userName = null, string? password = null, int? port = null);

public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistributedApplicationBuilder builder, string name, IResourceBuilder<ParameterResource> userNameParameter, IResourceBuilder<ParameterResource> passwordParameter, int? port = null);
```

This would then support all 3 scenarios from the issue:

1. Picking a username and password (using Parameter resources) - 2nd method
1. Defaulting a username and randomly generating a password - 1st method with `null, null`
1. Picking a username and generating a password - 1st method with `"username", null`

@davidfowl - for that last scenario, where the dev picks a user name, do you expect that user name to go into the manifest?